[bug] could not compile `tauri`

### Describe the bug

i can't compile tauri, i think it's related to the latest release

commit 7f885bd5edf3e179ddf0b674c3ea7a96e26ce39e

### Reproduction

_No response_

### Expected behavior

_No response_

### Full `tauri info` output

```text
[✔] Environment
    - OS: Windows 10.0.19045 X64
    ✔ WebView2: 124.0.2478.80
    ✔ MSVC:
        - Visual Studio Build Tools 2019
        - Visual Studio Community 2022
    ✔ rustc: 1.78.0 (9b00956e5 2024-04-29)
    ✔ cargo: 1.78.0 (54d8815d0 2024-03-26)
    ✔ rustup: 1.27.1 (54dd3d00f 2024-04-24)
    ✔ Rust toolchain: stable-x86_64-pc-windows-msvc (default)
    - node: 20.12.2
    - pnpm: 8.15.1
    - yarn: 1.22.19
    - npm: 10.5.0

[-] Packages
    - tauri [RUST]: 1.6.4
    - tauri-build [RUST]: 1.5.2
    - wry [RUST]: 0.24.9
    - tao [RUST]: 0.16.9
    - @tauri-apps/api [NPM]: 1.5.5
    - @tauri-apps/cli [NPM]: 1.5.13

[-] App
    - build-type: bundle
    - CSP: unset
    - distDir: ../dist
    - devPath: http://localhost:1420/
    - framework: Vue.js
    - bundler: Vite
```


### Stack trace

```text
Compiling tauri v1.6.4
error[E0412]: cannot find type `Command` in module `crate::api::process`
   --> C:\Users\admin\.cargo\registry\src\index.crates.io-6f17d22bba15001f\tauri-1.6.4\src\endpoints\shell.rs:230:41
    |
230 | ) -> super::Result<crate::api::process::Command> {
    |                                         ^^^^^^^ not found in `crate::api::process`
    |
help: consider importing this struct
    |
7   + use std::process::Command;
    |
help: if you import `Command`, refer to it directly
    |
230 - ) -> super::Result<crate::api::process::Command> {
230 + ) -> super::Result<Command> {
    |

   --> C:\Users\admin\.cargo\registry\src\index.crates.io-6f17d22bba15001f\tauri-1.6.4\src\endpoints\shell.rs:290:50
    |
290 |     if let Some(encoding) = crate::api::process::Encoding::for_label(encoding.as_bytes()) {
    |                                                  ^^^^^^^^ could not find `Encoding` in `process`
    |
help: consider importing this struct
    |
    |
help: if you import `Encoding`, refer to it directly
    |
290 -     if let Some(encoding) = crate::api::process::Encoding::for_label(encoding.as_bytes()) {
290 +     if let Some(encoding) = Encoding::for_label(encoding.as_bytes()) {
    |

Some errors have detailed explanations: E0412, E0433.
For more information about an error, try `rustc --explain E0412`.
error: could not compile `tauri` (lib) due to 2 previous errors
```


### Additional context

_No response_

--- a/.changes/shell-execute.md
+++ b/.changes/shell-execute.md
@@ -1,0 +1,5 @@
+---
+"tauri": "patch:bug"
+---
+
+Fix compilation error when `shell-execute` or `shell-sidecar` features are not active

--- a/core/tauri/src/endpoints/shell.rs
+++ b/core/tauri/src/endpoints/shell.rs
@@ -110,7 +110,8 @@ impl Cmd {
     let encoding = options
       .encoding
       .as_ref()
-      .and_then(|encoding| crate::api::process::Encoding::for_label(encoding.as_bytes()));
+      .and_then(|encoding| encoding_rs::Encoding::for_label(encoding.as_bytes()));
+
     let command = prepare_cmd(&context, &program, args, options)?;
 
     let mut command: std::process::Command = command.into();
@@ -222,6 +223,7 @@ impl Cmd {
   }
 }
 
+#[cfg(any(shell_execute, shell_sidecar))]
 fn prepare_cmd<R: Runtime>(
   context: &InvokeContext<R>,
   program: &String,
@@ -287,7 +289,7 @@ fn prepare_cmd<R: Runtime>(
     command = command.env_clear();
   }
   if let Some(encoding) = &options.encoding {
-    if let Some(encoding) = crate::api::process::Encoding::for_label(encoding.as_bytes()) {
+    if let Some(encoding) = encoding_rs::Encoding::for_label(encoding.as_bytes()) {
       command = command.encoding(encoding);
     } else {
       return Err(anyhow::anyhow!(format!("unknown encoding {encoding}")));

